File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int l=0, r=heights.length-1;
        int max = -1;
        while(l<r){
            max = Math.max(Math.min(heights[l],heights[r])* (r-l), max) ;
            if(heights[l]<=heights[r]){
                l+=1;
            }else{
                r-=1;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-0.py
class Solution:
    def climbStairs(self, n: int) -> int:
        # c(n) = 1, c(n-1)
        # c(n) = 2, c(n-2)
        # c(n-1) = 1, c(n-2)
        # c(n-1) = 2, c(n-3)
        if n==1:
            return 1
        elif n==2:
            return 2
        return max(1 + self.climbStairs(n-1), 2 + self.climbStairs(n-2))
